projects/npm-tools/packages/npm-scripts/src/scripts/types.js

```diff
@@ -9,10 +9,40 @@ const getTypeScriptBuildOrder = require('../typescript/getTypeScriptBuildOrder')
 const getTypeScriptDependencyGraph = require('../typescript/getTypeScriptDependencyGraph');
 const runTSC = require('../typescript/runTSC');
 const findRoot = require('../utils/findRoot');
+const getLiferayWorkingBranch = require('../utils/getLiferayWorkingBranch');
 const getMergedConfig = require('../utils/getMergedConfig');
+const git = require('../utils/git');
 const log = require('../utils/log');
 
-async function types() {
+function tsDiffFromWorkingBranch() {
+	const upstream = getLiferayWorkingBranch();
+
+	const mergeBase = git('merge-base', 'HEAD', upstream);
```

Comment:
Isn't this going to be `master` always? :thinking: 

Under which circumstance would it be something different? :thinking:

Reply:
It's almost always master, but it depends if someone sets `git.working.branch.name` in their portal.properties

Reply:
Ah, true :+1: